airflow/serialization/serialized_objects.py

```diff
@@ -623,7 +647,7 @@ def deserialize_operator(cls, encoded_op: Dict[str, Any]) -> BaseOperator:
 
         for k, v in encoded_op.items():
 
-            if k == "_downstream_task_ids":
+            if k in {"_downstream_task_ids", "downstream_task_ids"}:
```

Comment:
Maybe we should rename `downstream_task_ids` to `_downstream_task_ids` instead? This feels a bit awkward.